src/core/qgsmapsettings.cpp

```diff
@@ -641,3 +676,79 @@ void QgsMapSettings::writeXML( QDomNode& theNode, QDomDocument& theDoc )
 
   mDatumTransformStore.writeXML( theNode, theDoc );
 }
+
+void QgsMapSettings::updateDatumTransformEntries()
+{
+  if ( !hasCrsTransformEnabled() )
+    return;
+
+  QString destAuthId = destinationCrs().authid();
+  Q_FOREACH ( const QString& layerID, layers() )
+  {
+    QgsMapLayer* layer = QgsMapLayerRegistry::instance()->mapLayer( layerID );
+    if ( !layer )
+      continue;
+
+    QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( layer );
+    if ( vl && vl->geometryType() == QGis::NoGeometry )
+      continue;
+
+    // if there are more options, ask the user which datum transform to use
+    if ( !datumTransformStore().hasEntryForLayer( layer ) )
+    {
+      getDatumTransformInfo( layer, layer->crs().authid(), destAuthId );
+    }
+  }
+}
+
+void QgsMapSettings::getDatumTransformInfo( const QgsMapLayer* ml, const QString& srcAuthId, const QString& destAuthId )
+{
+  if ( !ml )
+  {
+    return;
+  }
+
+  //check if default datum transformation is available in environment variable
+  const char* envChar = getenv( "DEFAULT_DATUM_TRANSFORM" );
+  if ( envChar )
+  {
+    QString envString( envChar );
+    QStringList transformSplit = envString.split( ";" );
+    for ( int i = 0; i < transformSplit.size(); ++i )
+    {
+      QStringList slashSplit = transformSplit.at( i ).split( "/" );
+      if ( slashSplit.size() < 4 )
+      {
+        continue;
+      }
+
+      if ( slashSplit.at( 0 ) == srcAuthId && slashSplit.at( 1 ) == destAuthId )
+      {
+        datumTransformStore().addEntry( ml->id(), srcAuthId, destAuthId, slashSplit.at( 2 ).toInt(), slashSplit.at( 3 ).toInt() );
+        return;
+      }
+    }
+  }
+
+  //check if default datum transformation is available in settings
+  QSettings s;
```

Comment:
IMO `QgsMapSettings` shouldn't reach out to settings for things.. This makes it hard to use in standalone applications.   I'm not sure you would need a signal here.  Could it just be done with a different method `tranformRequired` that gets checked before doing this?